crates/database/src/alloydb.rs

```diff
@@ -78,10 +80,15 @@ impl<N: Network, P: Provider<N>> DatabaseAsyncRef for AlloyDB<N, P> {
 
         let balance = balance?;
         let code = Bytecode::new_raw(code?.0.into());
-        let code_hash = code.hash_slow();
         let nonce = nonce?;
+        let code_hash = code.hash_slow();
 
-        Ok(Some(AccountInfo::new(balance, nonce, code_hash, code)))
+        let empty_acc = SpecId::is_enabled_in(self.spec_id, SpecId::SPURIOUS_DRAGON);
+        if empty_acc && nonce == 0 && balance.is_zero() && code.is_empty() {
```

Comment:
This is not a solution for it.

I don't think there is right fix for this. You would need to fetch a additional info that is not available and that is if account exist or not. As of now with just `nonce == 0 && balance.is_zero() && code.is_empty()` we don't know if account is empty (pre state clear empty) or not.